config/manager/network_policy.yaml

```diff
@@ -0,0 +1,21 @@
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-ingress-https-only-and-block-egress
```

Comment:
The name concerns me. The other PR adds more things to the network policy... I don't think we can have every single thing the policy does in the name. IMO, we should rely on inspecting the policy to see what it does, and we can use the commit history to see why certain aspects were added. Maybe it should be something like `controller-manager` or something so we can easily see it is for the manager pod